From 9d0a5f0c6edacb7e2d2af84d0e63a7f0d4643475 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:13:41 -0400
Subject: [alsa] Enable CONFIG_ZONE_DMA for smaller PCI DMA masks

Message-id: <140972823573-32-patchset-jkysela@redhat.com>
Patchwork-id: 90732
O-Subject: [RHEL 7.1 PATCH 031/442] ALSA: Enable CONFIG_ZONE_DMA for smaller PCI DMA masks
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The PCI devices with DMA masks smaller than 32bit should enable
CONFIG_ZONE_DMA.  Since the recent change of page allocator, page
allocations via dma_alloc_coherent() with the limited DMA mask bits
may fail more frequently, ended up with no available buffers, when
CONFIG_ZONE_DMA isn't enabled.  With CONFIG_ZONE_DMA, the system has
much more chance to obtain such pages.

Bugzilla: https://bugzilla.kernel.org/show_bug.cgi?id=68221
Cc: <stable@vger.kernel.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 80ab8eae70e51d578ebbeb228e0f7a562471b8b7)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/Kconfig b/sound/pci/Kconfig
index fe6fa93..daac7c7 100644
--- a/sound/pci/Kconfig
+++ b/sound/pci/Kconfig
@@ -30,6 +30,7 @@ config SND_ALS300
  select SND_PCM
  select SND_AC97_CODEC
  select SND_OPL3_LIB
+ select ZONE_DMA
  help
    Say 'Y' or 'M' to include support for Avance Logic ALS300/ALS300+
 
@@ -54,6 +55,7 @@ config SND_ALI5451
  tristate "ALi M5451 PCI Audio Controller"
  select SND_MPU401_UART
  select SND_AC97_CODEC
+ select ZONE_DMA
  help
    Say Y here to include support for the integrated AC97 sound
    device on motherboards using the ALi M5451 Audio Controller
@@ -158,6 +160,7 @@ config SND_AZT3328
  select SND_PCM
  select SND_RAWMIDI
  select SND_AC97_CODEC
+ select ZONE_DMA
  help
    Say Y here to include support for Aztech AZF3328 (PCI168)
    soundcards.
@@ -463,6 +466,7 @@ config SND_EMU10K1
  select SND_HWDEP
  select SND_RAWMIDI
  select SND_AC97_CODEC
+ select ZONE_DMA
  help
    Say Y to include support for Sound Blaster PCI 512, Live!,
    Audigy and E-mu APS (partially supported) soundcards.
@@ -478,6 +482,7 @@ config SND_EMU10K1X
  tristate "Emu10k1X (Dell OEM Version)"
  select SND_AC97_CODEC
  select SND_RAWMIDI
+ select ZONE_DMA
  help
    Say Y here to include support for the Dell OEM version of the
    Sound Blaster Live!.
@@ -511,6 +516,7 @@ config SND_ES1938
  select SND_OPL3_LIB
  select SND_MPU401_UART
  select SND_AC97_CODEC
+ select ZONE_DMA
  help
    Say Y here to include support for soundcards based on ESS Solo-1
    (ES1938, ES1946, ES1969) chips.
@@ -522,6 +528,7 @@ config SND_ES1968
  tristate "ESS ES1968/1978 (Maestro-1/2/2E)"
  select SND_MPU401_UART
  select SND_AC97_CODEC
+ select ZONE_DMA
  help
    Say Y here to include support for soundcards based on ESS Maestro
    1/2/2E chips.
@@ -603,6 +610,7 @@ config SND_ICE1712
  select SND_MPU401_UART
  select SND_AC97_CODEC
  select BITREVERSE
+ select ZONE_DMA
  help
    Say Y here to include support for soundcards based on the
    ICE1712 (Envy24) chip.
@@ -690,6 +698,7 @@ config SND_LX6464ES
 config SND_MAESTRO3
  tristate "ESS Allegro/Maestro3"
  select SND_AC97_CODEC
+ select ZONE_DMA
  help
    Say Y here to include support for soundcards based on ESS Maestro 3
    (Allegro) chips.
@@ -786,6 +795,7 @@ config SND_SIS7019
  tristate "SiS 7019 Audio Accelerator"
  depends on X86 && !X86_64
  select SND_AC97_CODEC
+ select ZONE_DMA
  help
    Say Y here to include support for the SiS 7019 Audio Accelerator.
 
@@ -797,6 +807,7 @@ config SND_SONICVIBES
  select SND_OPL3_LIB
  select SND_MPU401_UART
  select SND_AC97_CODEC
+ select ZONE_DMA
  help
    Say Y here to include support for soundcards based on the S3
    SonicVibes chip.
@@ -808,6 +819,7 @@ config SND_TRIDENT
  tristate "Trident 4D-Wave DX/NX; SiS 7018"
  select SND_MPU401_UART
  select SND_AC97_CODEC
+ select ZONE_DMA
  help
    Say Y here to include support for soundcards based on Trident
    4D-Wave DX/NX or SiS 7018 chips.
-- 
1.7.1